### PR TITLE
feat(probe): Allow probing a specific address for cameras

### PR DESCRIPTION
**Pull request by GeodeLX**
The current implementation of node-onvif sends a multicast packet,
which is sent on only one interface.  For systems with multiple
network interfaces, we need to allow probing on specific interface
addresses.  This change allows the caller to specify an address
in the startProbe method.  If no address is given, the multicast
address is used (so no change to existing functionality).  If an
address is given, the probe packet is sent to that address.

### CODE REVIEW

**Comment by GeodeLX** [comment]
@nanpx @bschmickley @psevery 
Please review and confirm or correct.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-onvif",
-  "version": "0.1.7",
+  "version": "0.1.8",
```
Comment (nanpx) [inline package.json]: A feature would bump the semver to 0.2.0
```suggestion
  "version": "0.2.0",
```

File: lib/node-onvif.js [hunk 2]
```diff
@@ -52,9 +52,9 @@ Onvif.prototype.startDiscovery = function(callback) {
 };
 
 /* ------------------------------------------------------------------
-* Method: startProbe([callback])
+* Method: startProbe([address, callback])
 * ---------------------------------------------------------------- */
-Onvif.prototype.startProbe = function(callback) {
+Onvif.prototype.startProbe = function(address, callback) {
```
Comment (nanpx) [inline lib/node-onvif.js]: While traditionally the callback is the last argument, I think you should make it the second argument. If you look at `_execCallback` it's done in the same style - https://github.com/LGSInnovations/node-onvif/pull/1/files#diff-630a7d91ebc54e8fae0380339b58c5d4R164 - Also, you would do the same in `_sendProbe` and can eliminate the `typeof address === function` logic.
```suggestion
Onvif.prototype.startProbe = function(callback, address) {
```

File: lib/node-onvif.js [hunk 3]
```diff
@@ -167,7 +167,14 @@ Onvif.prototype._execCallback = function(callback, arg1, arg2) {
...
 
-Onvif.prototype._sendProbe = function(callback) {
+Onvif.prototype._sendProbe = function(address, callback) {
+	if (typeof address === 'function') {
+		callback = address;
+		address = undefined;
+	}
+	if (!address) {
+		address = this._MULTICAST_ADDRESS;
+	}
```
Comment (nanpx) [inline lib/node-onvif.js:173]:
```suggestion
Onvif.prototype._sendProbe = function(callback, address) {
	if (!address) {
		address = this._MULTICAST_ADDRESS;
	}
```

**Review by nanpx — CHANGES_REQUESTED** [review]

File: lib/node-onvif.js [hunk 4]
```diff
@@ -52,9 +52,9 @@ Onvif.prototype.startDiscovery = function(callback) {
 };
 
 /* ------------------------------------------------------------------
-* Method: startProbe([callback])
+* Method: startProbe([address, callback])
```
Comment (nanpx) [inline lib/node-onvif.js:55]: You'll want to update this too

File: lib/node-onvif.js [hunk 5]
```diff
@@ -167,7 +167,17 @@ Onvif.prototype._execCallback = function(callback, arg1, arg2) {
...
+	if (typeof address === 'function') {
+		callback = address;
+		address = undefined;
+	}
+	if (!address) {
+		address = this._MULTICAST_ADDRESS;
+Onvif.prototype._sendProbe = function(callback, address) {
+	if (!address) {
+		address = this._MULTICAST_ADDRESS;
+	}
```
Comment (nanpx) [inline lib/node-onvif.js:173]: Looks like this got a little messed up, I think you should be good to delete 170-176

**Review by nanpx — CHANGES_REQUESTED** [review]

**Pull request closed without merging**